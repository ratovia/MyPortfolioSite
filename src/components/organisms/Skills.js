import React from "react"
import { CircleImageIcon } from "../atoms/CircleImageIcon"
import { FrederickaH2 } from "../atoms/FrederickaText"
import styled from "styled-components"

const SkillsArea = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: 60px 30px;
`

const TitleText = styled.div`
  align-self: flex-end;
  margin-bottom: 60px;
`

const SkillListTop = styled.div`
  align-self: flex-end;
  display: flex;
  justify-content: space-around;
`
const SkillListMiddle = styled.div`
  align-self: center;
  display: flex;
  justify-content: space-around;
`
const SkillListBottom = styled.div`
  align-self: flex-start;
  display: flex;
  justify-content: space-around;
`

const Container = styled.div`
  padding: 10px 40px;
`
export const Skills = ({ skills }) => {
  return (
    <SkillsArea>
      <TitleText>
        <FrederickaH2 label="skills" color="white"></FrederickaH2>
      </TitleText>
      <SkillListTop>
        {skills.slice(0, 3).map(skill => {
          return (
            <Container key={skill.name}>
              <CircleImageIcon
                imageUrl={skill.imageUrl}
                size="medium"
              ></CircleImageIcon>
            </Container>
          )
        })}
      </SkillListTop>
      <SkillListMiddle>
        {skills.slice(3, 6).map(skill => {
          return (
            <Container key={skill.name}>
              <CircleImageIcon
                imageUrl={skill.imageUrl}
                size="medium"
              ></CircleImageIcon>
            </Container>
          )
        })}
      </SkillListMiddle>
      <SkillListBottom>
        {skills.slice(6, 9).map(skill => {
          return (
            <Container key={skill.name}>
              <CircleImageIcon
                imageUrl={skill.imageUrl}
                size="medium"
              ></CircleImageIcon>
            </Container>
          )
        })}
      </SkillListBottom>
    </SkillsArea>
  )
}
