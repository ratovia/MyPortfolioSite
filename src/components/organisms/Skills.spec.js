import React from "react"
import renderer from "react-test-renderer"
import backgroundImg from "../../stories/assets/bg.jpg"

import { Skills } from "./Skills"
describe("Skills", () => {
  it("renders correctly", () => {
    const skills = [
      {
        name: "Ruby on Rails",
        imageUrl: backgroundImg,
      },
      {
        name: "HTML/CSS",
        imageUrl: backgroundImg,
      },
      {
        name: "JavaScript",
        imageUrl: backgroundImg,
      },
      {
        name: "Python",
        imageUrl: backgroundImg,
      },
      {
        name: "PHP",
        imageUrl: backgroundImg,
      },
      {
        name: "AWS",
        imageUrl: backgroundImg,
      },
      {
        name: "3DCG",
        imageUrl: backgroundImg,
      },
      {
        name: "linux",
        imageUrl: backgroundImg,
      },
      {
        name: "React",
        imageUrl: backgroundImg,
      },
    ]
    const tree = renderer.create(<Skills skills={skills}></Skills>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
