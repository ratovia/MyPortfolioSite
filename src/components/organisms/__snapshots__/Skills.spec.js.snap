// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders correctly 1`] = `
<div
  className="sc-jRQAMF kLBZMp"
>
  <div
    className="sc-gKckTs MMGpI"
  >
    <h2
      className="sc-hKwCoD ciajrW"
    >
      <p
        className="sc-gsDJrp fFMVcA"
        color="white"
      >
        skills
      </p>
    </h2>
  </div>
  <div
    className="sc-iCfLBT cYYPVt"
  >
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
  </div>
  <div
    className="sc-furvIG dPzpIM"
  >
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
  </div>
  <div
    className="sc-pVTma iGbuvK"
  >
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
    <div
      className="sc-jrQzUz gtKmVj"
    >
      <img
        alt="アイコン"
        className="sc-bdvvaa hVPPTo"
        height="200px"
        src="test-file-stub"
        width="200px"
      />
    </div>
  </div>
</div>
`;
